examples_tests/44.LevelCurveExtraction/main.cpp

```diff
@@ -30,40 +30,118 @@ void main()
 
 const char* geometryShaderCode = R"===(
 #version 440 core
+
+#define FLT_MAX 3.402823466e+38
 struct DrawIndirectArrays_t
 {
-	uint  count;
-	uint  instanceCount;
-	uint  first;
-	uint  baseInstance;
+    uint  count;
+    uint  instanceCount;
+    uint  first;
+    uint  baseInstance;
 };
-layout (triangles) in;
-layout (triangle_strip, max_vertices = 3) out;
-layout (location = 0) in vec3 LocalPos[];
-layout (location = 1) in vec3 ViewPos[];
-layout (location = 2) in vec3 Normal[];
-layout (location = 0) out vec3 fragLocalPos;
-layout (location = 1) out vec3 fragViewPos;
-layout (location = 2) out vec3 fragNormal;
+
+layout(triangles) in;
+layout(triangle_strip, max_vertices = 3) out;
+layout(location = 0) in vec3 LocalPos[];
+layout(location = 1) in vec3 ViewPos[];
+layout(location = 2) in vec3 Normal[];
+layout(location = 0) out vec3 fragLocalPos;
+layout(location = 1) out vec3 fragViewPos;
+layout(location = 2) out vec3 fragNormal;
 #ifndef _NO_UV
-layout (location = 3) in vec2 UV[];
-layout (location = 3) out vec2 fragUV;
+layout(location = 3) in vec2 UV[];
+layout(location = 3) out vec2 fragUV;
 #endif
 
-layout(set=0, binding=0) coherent buffer LineCount
+layout(set = 0, binding = 0) coherent buffer LineCount
 {
-  DrawIndirectArrays_t lineDraw;
+    DrawIndirectArrays_t lineDraw;
 };
-layout(set=0, binding=1) writeonly buffer Lines
+layout(set = 0, binding = 1) writeonly buffer Lines
 {
-  float linePoints[]; // 6 floats decribe a line, 3d start, 3d end
+    float linePoints[]; // 6 floats decribe a line, 3d start, 3d end
 };
 
-void main() {    
+void main() {
+    const float levelPlanesDistance= 10.0;
+    const vec3 levelPlaneNormal = vec3(0.0,1.0,0.0);   
+    uint numHorLines;
+    float maxLevel = -FLT_MAX;
+    float minLevel = FLT_MAX;
     uint i;
-    vec3 end = vec3(0,0,0);
-    vec3 start = LocalPos[0];
-    for(i = 0;i < gl_in.length();i++)
+    float vertexPlaneDistance[3];
+    for (i = 0; i < 3; i++)
+    {
+        vertexPlaneDistance[i] =dot(levelPlaneNormal, LocalPos[i]);
+        maxLevel = max(maxLevel,vertexPlaneDistance[i]);
+        minLevel = min(minLevel,vertexPlaneDistance[i]);
+    }
+    int sharedMxVtx = 0,sharedMnVtx = 0;
+    for (i = 0; i < 3; i++)
+    {
+        if (vertexPlaneDistance[i] == maxLevel)
+			sharedMxVtx++;
+		if (vertexPlaneDistance[i] == minLevel)
+			sharedMnVtx++;
+    }
+    if(sharedMnVtx <2)
+		minLevel += 0.001f;
+	if(sharedMxVtx < 2)
+		maxLevel -= 0.001f;
```

Comment:
do tie breaking, so one triangle will have to create this line (out of the two that share the edge)... my suggestion is to only do this check for the top edge.